cli/listener/src/main.rs

```diff
@@ -213,44 +223,332 @@ pub fn get_apy_for_period(
     // Ok(Some(1.0))
 }
 
-pub fn insert_price(conn: &Connection, exchange_rate: ExchangeRate) -> rusqlite::Result<()> {
+pub fn insert_price(conn: &Connection, exchange_rate: &ExchangeRate) -> rusqlite::Result<()> {
     conn.execute("INSERT INTO exchange_rate (timestamp, slot, epoch, pool, price_lamports_numerator, price_lamports_denominator) VALUES (?1, ?2, ?3, ?4, ?5, ?6)", 
     params![exchange_rate.timestamp.to_string(), exchange_rate.slot, exchange_rate.epoch, exchange_rate.pool,
```

Comment:
I noticed when inspecting the database that it produces strings of the form `2022-02-01 11:58:39.923080375 UTC`. That’s not ISO-8601. We should use [`to_rfc3339`](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html#method.to_rfc3339) to get an ISO-8601-compatible format.

I suppose it also needs a change when we read the value to use `parse_from_rfc3339`. The reason to use ISO-8601 is that it is widely compatible with other applications, (e.g. Python has `datetime.isoformat()` and `fromisoformat()`), and SQLite’s built-in date functions can handle it as well.